oximeter/types/src/histogram.rs

```diff
@@ -1029,8 +1029,10 @@ where
             return Err(QuantizationError::InvalidSteps);
         }
 
-        // The highest power must be representable in the target type.
-        if self.checked_pow(hi.into()).is_none() {
+        // The highest power must be representable in the target type. Note that
+        // we have to convert to that target type _before_ doing this check.
+        let base = <u64 as From<Self>>::from(*self);
```

Comment:
This is not correct, it's checking in `u64`-land, needs to try to convert to the bin type `T`

Reply:
Fixed in d0ccdb0